pkg/infrastructure/aws/vpc.go

```diff
@@ -181,13 +202,31 @@ func createVPCResources(ctx context.Context, logger logrus.FieldLogger, ec2Clien
 				tableIdx = 0
 			}
 			table := privateRouteTables[tableIdx]
-			subnetID := subnetZoneMap[zone]
+			subnetID := privateZoneMap[zone]
 			if err := addSubnetToRouteTable(ctx, logger, ec2Client, table, subnetID); err != nil {
 				return nil, fmt.Errorf("failed to associate edge subnet (%s) to route table (%s): %w", aws.StringValue(subnetID), aws.StringValue(table.RouteTableId), err)
 			}
+
+			// Associate public subnet with either the public or carrier route
+			// table depending on the zone type
+			table = publicRouteTable
+			tableType := "public"
+			if zoneType, ok := vpcInput.edgeZonesType[zone]; ok && zoneType == awstypes.WavelengthZoneType {
+				table = carrierTable
+				tableType = "carrier"
+			}
+			publicSubnetID := publicZoneMap[zone]
```

Comment:
Are we handling private installations in the altinfra version / this flow? I am also have doubts how the private/disconnected installations are handled in TF version as long we are also [checking only the zone length](https://github.com/openshift/installer/blob/1709379a76141e94113382f5514be6a4b5718e4b/data/data/aws/cluster/vpc/vpc-public.tf#L50-L51)

Reply:
I don't know how BYO VPCs related to the edge/wavelength zones though. Depending on what's expected, we might be missing a few bits.

Reply:
@r4f4 , you are correct. I revisit the restricted[1]/private[2] and all uses BYO VPC path which will not touch that part. BYO VPC for WL follows the same pattern.
[1] https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-restricted-networks-aws-installer-provisioned.html
[2] https://docs.openshift.com/container-platform/4.14/installing/installing_aws/installing-aws-private.html